From e7c9800c4176b901dcb2cc7f52e53bb1b529ab03 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mathieu=20Dupr=C3=A9?= <mathieu.dupre@savoirfairelinux.com>
Date: Thu, 11 May 2023 11:18:42 +0200
Subject: [PATCH 1/3] grub-efi: do not deploy modules if they're not installed
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Check that the folder actually exists in the path prior attempting to
deploy it.

Signed-off-by: Sébastien LE STUM <sebastien.le-stum@savoirfairelinux.com>
Signed-off-by: Mathieu Dupré <mathieu.dupre@savoirfairelinux.com>
---
 .../recipes-bsp/grub/grub-efi-efi-secure-boot.inc             | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc b/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc
index 8fbb75c..82cb5d7 100644
--- a/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc
+++ b/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc
@@ -193,7 +193,9 @@ do_deploy:append:class-target() {
 
     install -d "${DEPLOYDIR}/efi-unsigned"
     install -m 0644 "${B}/${GRUB_IMAGE}" "${DEPLOYDIR}/efi-unsigned"
-    PSEUDO_DISABLED=1 cp -af "${D}${EFI_BOOT_PATH}/${GRUB_TARGET}-efi" "${DEPLOYDIR}/efi-unsigned"
+    if [ -d "${D}${EFI_BOOT_PATH}/${GRUB_TARGET}-efi" ]; then
+        PSEUDO_DISABLED=1 cp -af "${D}${EFI_BOOT_PATH}/${GRUB_TARGET}-efi" "${DEPLOYDIR}/efi-unsigned"
+    fi
 }
 
 FILES:${PN} += "${EFI_BOOT_PATH}"
-- 
2.25.1

